### PR TITLE
[DOCS] Remove unneeded redirects

### PR DESCRIPTION
**Pull request by jrodewig**
The `docs/reference/redirects.asciidoc` file stores a list of relocated or deleted pages for the Elasticsearch Reference documentation.

This prunes several older redirects that are no longer needed and don't require work to fix broken links in other repositories. Where needed, it also updates anchors within the Elasticsearch docs.

I plan to remove more in future PRs, but most of those require fixing would-be-broken links in other repositories.

### CODE REVIEW

File: docs/reference/ml/anomaly-detection/apis/ml-api.asciidoc [hunk 1]
```diff
@@ -4,7 +4,7 @@
 == {ml-cap} {anomaly-detect} APIs
 
 You can use the following APIs to perform {ml} {anomaly-detect} activities. See
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/ml-api.asciidoc]: > See ... advanced job configuration options in Kibana.

We're removing the separate definitions pages for the ML APIs, so I think it's better to just remove this sentence.  Thanks for catching this!

Comment (jrodewig) [reply docs/reference/ml/anomaly-detection/apis/ml-api.asciidoc]: Removed with 3b0f6f383cda1bac6793a28d5bbca60020a15a48. Thanks @lcawl!

**Review by lcawl — APPROVED** [review]
LGTM, thanks!

**Comment by jrodewig** [comment]
### Backport commits
master: e8a6d4a 
7.x: ef467cc6f5807855b12ff8756b1e07b8cbdf0910

**Pull request merged**